### ISSUE TITLE
[QUESTION]Problems with version upgrades

### ISSUE THREAD

**Issue by zyang6** [NVIDIA/Megatron-LM#1659]
Why is a memory address alignment operation performed here during the upgrade from version 060 to 080? Does it have any impact on the entire code? After this upgrade, my loss curve has changed. 

CODE：megatron/core/distributed/param_and_grad_buffer.py:303

        for param in params[::-1]:
            # Iterate through parameters in reverse order to roughly follow backprop order,
            # and skip parameters that don't require gradients.
            if not param.requires_grad:
                continue
            this_numel = param.data.nelement()
            data_start_index = _pad_start_of_param_if_needed(data_start_index)
            data_end_index = data_start_index + this_numel

**Comment by sbhavani** [comment]
@zyang6 sorry for the late response. This alignment is intentional for the distributed optimizer and helps avoid incompatible or inefficient cuBLAS kernel selection; it should not change parameter values, though small numerical differences are possible. 

If you are still facing loss/accuracy issues, please share a minimal repro and exact version/configuration details.
